docs/src/configuration/app/source-operations.md

```diff
@@ -25,42 +40,160 @@ source:
+
+1. The current environment HEAD commit is checked out in Git.
+   It doesn't have any remotes or tags defined in the project.
+   It only has the current environment branch.
+2. Sequentially, for each app that has an operation with this name in its configuration,
+   the operation command is run in the app container.
+   The container isn't part of the environment's runtime cluster
+   and doesn't require that the environment is running.
+
+   The environment has all of the variables normally available during the build phase.
+   These may be optionally overridden by the variables specified when the operation is run.
+3. If any new commits were created, they're pushed to the repository and the normal build process is triggered.
+
+   If multiple apps in a single project both result in a new commit,
+   there are two distinct commits in the Git history but only a single new build process.
+
+## Use variables in your source operations
+
+You can add variables to the environment of the source operation.
+
+They're interpreted the same way as any other [variable](../../development/variables/_index.md)
+already set in your project.
+That means you need an `env:` prefix to expose them as a Unix environment variable.
+
 They can then be referenced by the source operation like any other variable.
 
+Say you wanted to have a `FILE` variable available with the value `example.txt`.
+Run the operation with the variable:
+
+{{< codetabs >}}
+
+---
+title=In the Console
+file=none
+highlight=false
+---
+
+<!--This is in HTML to get the icon not to break the list. -->
+<ol>
+  <li>Navigate to the environment where you want to run the operation.</li>
+  <li>Click {{< icon more >}} <strong>More</strong>.</li>
+  <li>Click <strong>Run source operation</strong>.</li>
+  <li>Select the operation you want to run.
+  <li>Under <strong>Add/override variables</strong>, put <code>FILE</code> as the <strong>Variable name</strong> and <code>example.txt</code> as the <strong>Value</strong>. The variable is automatically prefixed with <code>env:</code>.</li>
+  <li>Click <strong>Run</strong>.</li>
+</ol>
+
+<--->
+---
+title=Using the CLI
+file=none
+highlight=false
+---
+
 ```bash
-platform source-operation:run update --variable env:FOO=bar --variable env:BAZ=beep
+platform source-operation:run update --variable env:FILE="example.txt"
+```
+
+{{< /codetabs >}}
+
+## Source integrations
+
+If your project is using a [source integration](../../integrations/source/_index.md),
+any new commits resulting from a source operation are first pushed to your external Git repository.
+Then the source integration pushes those commits to Platform.sh and redeploys the environment.
+
+When using a source integration,
+you can't run source operations on environments created from pull or merge requests created on the external repository.
+
+If you try running a source operation on a non-supported environment, you see the following error:
+
+```text
```

Comment:
The error message is unclear and will probably be improved in an upcoming version. [More context in (1)](https://github.com/orgs/platformsh/projects/3/views/1)

Reply:
Should I add the example comment for clarity?

Reply:
I would document the existing state and then change anything once that change comes.